client/components/userLineChart.jsx

```diff
@@ -0,0 +1,65 @@
+import { Line } from 'react-chartjs-2';
+import React from 'react';
+// eslint-disable-next-line no-unused-vars
+import Chart from 'chart.js/auto';
+
+export default class LineChart extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      chartData: {},
+      isLoading: true
+    };
+    this.getChartData = this.getChartData.bind(this);
+  }
+
+  getChartData() {
+    fetch('/api/entries')
+      .then(res => res.json())
+      .then(data => {
+        const chartData = data;
+        const dates = [];
+        const weights = [];
+        chartData.forEach(element => {
+          const { date } = element;
+          const modDate = date.split('T');
+          const finalDate = modDate[0];
+          const labelDate = finalDate.split('-').splice(1, 2).join('-');
+          dates.unshift(labelDate);
+          weights.unshift(element.weight);
```

Comment:
`.unshift()` is pretty computationally expensive. Instead, it's better to use `order by` in your SQL query. you can order by date and timestamp columns, just like numbers.